### PR TITLE
Set up codecov workflow and badges

### LINKED ISSUE #287: Add pypi badges and coverage badge
And include coverage page in docs

### PR DESCRIPTION
**Pull request by tsmbland**
# Description

Adds a step to the CI workflow to upload the coverage report to codecov, and link this page from the documentation.

NOTE: the workflow is currently failing, but we believe it will only work once it has been merged to main

This replaces a half-implemented workflow using the [mkdocs-coverage](https://pypi.org/project/mkdocs-coverage/) package which would be considerably more effort to set up properly. I also needed to change the coverage report from html to xml to get it working with codecov.

Also adds badges for codecov and PyPI to the readme

Fixes #287 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

**Comment by tsmbland** [comment]
@dalonsoa Please could you generate a codecov token and add to the repository secrets?

**Comment by dalonsoa** [comment]
Done. It should be available as `secrets.CODECOV_TOKEN`, I think.

**Comment by tsmbland** [comment]
Thanks, although it's not finding it for some reason...

**Review by dalonsoa — APPROVED** [review]
As discussed, the workflow is failing, but I think it is for good reasons and will work once merged into main. Let's see. We can always undo it if it does not work.

**Comment by tsmbland** [comment]
@dalonsoa You'l have to merge it for me as I'm blocked by branch-protection rules

**Comment by tsmbland** [comment]
Still not working 😭 😭 😭 😭

**Pull request merged**
